mne/minimum_norm/resolution_matrix.py

```diff
@@ -63,51 +67,236 @@ def make_inverse_resolution_matrix(forward, inverse_operator, method='dSPM',
     return resmat
 
 
-def _get_psf_ctf(resmat, src, idx, func='psf', norm=False):
-    """Get point-spread (PSFs) or cross-talk (CTFs) functions for vertices.
+@verbose
+def _get_psf_ctf(resmat, src, idx, func, mode, n_comp, norm, return_pca_vars,
+                 verbose=None):
+    """Get point-spread (PSFs) or cross-talk (CTFs) functions.
 
     Parameters
     ----------
     resmat : array, shape (n_dipoles, n_dipoles)
         Forward Operator.
     src : Source Space
         Source space used to compute resolution matrix.
-    idx : list of int
-        Vertex indices for which PSFs or CTFs to produce.
+    %(pctf_idx)s
     func : str ('psf' | 'ctf')
         Whether to produce PSFs or CTFs. Defaults to psf.
-    norm : bool
-        Whether to normalise to maximum across all PSFs and CTFs (default:
-        False).
+    %(pctf_mode)s
+    %(pctf_n_comp)s
+    %(pctf_norm)s
+    %(pctf_return_pca_vars)s
+    %(verbose)s
 
     Returns
     -------
-    stc: instance of SourceEstimate
-        PSFs or CTFs as an stc object.
+    %(pctf_stcs)s
+    %(pctf_pca_vars)s
     """
+    # check for consistencies in input parameters
+    _check_get_psf_ctf_params(mode, n_comp, return_pca_vars)
+
+    # backward compatibility
+    if norm is True:
+        norm = 'max'
+
+    # get relevant vertices in source space
+    verts_all = _vertices_for_get_psf_ctf(idx, src)
+
     # vertices used in forward and inverse operator
     vertno_lh = src[0]['vertno']
     vertno_rh = src[1]['vertno']
     vertno = [vertno_lh, vertno_rh]
 
-    # in everything below indices refer to columns
+    # the following will operate on columns of funcs
     if func == 'ctf':
         resmat = resmat.T
 
-    # column of resolution matrix
-    funcs = resmat[:, idx]
+    # Functions and variances per label
+    stcs = []
+    pca_vars = []
+
+    for verts in verts_all:
+
+        # get relevant PSFs or CTFs for specified vertices
+        funcs = resmat[:, verts]
+
+        # normalise PSFs/CTFs if requested
+        if norm is not None:
+
```

Comment:
I would make this whole function much more compact. No need to put spaces between every operation/clause